src/In.ProjectEKA.TMHHip/DataFlow/Collect.cs

```diff
@@ -0,0 +1,662 @@
+                observations.Add(observationRepresentation);
+                observations.Add(new OralCavityExaminationObservationRepresention(tongueResource));
+                observations.Add(new OralCavityExaminationObservationRepresention(buccalmcsaResource));
+            }
+
+            if (!observations.Any())
+            {
+                return Option.None<ObservationResponse>();
+            }
+
+            var oralCavityExmResponse = new ObservationResponse
+            {
+                Entry = observations,
+                Id = Uuid.Generate().Value.Split(":").Last(),
+                ResourceType = "Bundle",
+                Type = "collection"
+            };
+            return Option.Some(oralCavityExmResponse);
+        }
+
+        private async Task<PatientData> FetchPatientData(
+            DataRequest dataRequest)
+        {
+            try
+            {
+                LogDataRequest(dataRequest);
+                var caseId = dataRequest.CareContexts.First().PatientReference;
+                var request = new HttpRequestMessage(HttpMethod.Post, "https://tmc.gov.in/tmh_ncg_api/healthInfo")
+                {
+                    Content = new StringContent(
+                        JsonConvert.SerializeObject(new
+                        {
+                            HiTypes = dataRequest.HiType,
+                            StartDate = dataRequest.DateRange.From,
+                            EndDate = dataRequest.DateRange.To,
+                            CaseId = caseId
+                        }),
+                        Encoding.UTF8,
+                        "application/json")
+                };
+
+                var response = await client.SendAsync(request).ConfigureAwait(false);
+                var responseStream = await response.Content.ReadAsStreamAsync();
+                return await JsonSerializer.DeserializeAsync<PatientData>(responseStream);
+            }
+            catch (Exception e)
+            {
+                Log.Error(e.Message);
+                return null;
+            }
+        }
+
+        private static void LogDataRequest(DataRequest request)
+        {
+            var ccList = JsonConvert.SerializeObject(request.CareContexts);
+            var requestedHiTypes = string.Join(", ", request.HiType.Select(hiType => hiType.ToString()));
+            Log.Information("Data request received." +
+                            $" transactionId:{request.TransactionId} , " +
+                            $"CareContexts:{ccList}, " +
+                            $"HiTypes:{requestedHiTypes}," +
+                            $" From date:{request.DateRange.From}," +
+                            $" To date:{request.DateRange.To}, " +
+                            $"CallbackUrl:{request.DataPushUrl}");
+        }
+
+        private static string PdfToBase64(string pdfUrl)
+        {
+            string base64String;
+            using (WebClient client = new WebClient())
```

Comment:
Codacy found an issue: [Rename 'client' which hides the field with the same name.](https://app.codacy.com/gh/ProjectEKA/hip-service/pullRequest?prid=4940156)